### DISCUSSION TITLE
Possible to get the IP address in a RPC?

### DISCUSSION THREAD

**Discussion by pierroo** [supabase/supabase#27002] [Questions] [answered]
I see in my supabase logs for Postrges / api calls / auth the remote_address or ip / location etc,
is it possible to get those information from a rpc?

for example, I do a supabase.rpc('get_info')
=> from the supabase function I can get the user.id for example, any way to have the IP address?

**Comment by jacobrobertsbaca** [comment]
Bumping this

**Reply by GaryAustin1** [reply, marked as answer]
Yes you can get the ip address if the call is to the database.

You can use a test rpc/function to return all the available headers (return jsonb type).
```sql
BEGIN
return
 current_setting('request.headers', true);
END;
```
You would get specific headers with ` current_setting('request.headers', true)->>'headername'`

An example:
![image](https://github.com/user-attachments/assets/b1b1f874-dbf8-4cdd-8f92-759a15c8c67a)
